scalafix-core/shared/src/main/scala/scalafix/internal/rule/MissingFinal.scala

```diff
@@ -31,11 +31,11 @@ final case class MissingFinal(index: SemanticdbIndex)
 
     def addFinal(mods: Seq[Mod], d: Defn): Patch =
       (mods.find(!_.is[Mod.Annot]) match {
-        case Some(mod) => ctx.addLeft(mod, "final ")
+        case Some(mod) => ctx.addLeft(mod, "final ").atomic
         case None =>
           mods.lastOption match {
-            case Some(lastMod) => ctx.addRight(lastMod, " final")
-            case None => ctx.addLeft(d, "final ")
+            case Some(lastMod) => ctx.addRight(lastMod, " final").atomic
+            case None => ctx.addLeft(d, "final ").atomic
           }
       }).atomic
```

Comment:
this rule already emits atomic patches (line 40). In order to avoid redundancy, I'd keep it as is or only your changes if you believe they make it clearer